Simplify code, add a bit of new functionality.

fritzgrabo opened the pull request:
I created a trimmed down version of `helm-mt`to make it a more light-weight wrapper around `multi-term`s existing functionality.

~~Note that this is somewhat opinionated as it drops support for creation of terminals in `shell-mode`. I realize that this can be considered a show-stopper by some. That said, if `shell-mode` support is desired, shouldn't it rather go into `multi-term`?~~
- Use existing `multi-term-buffer-list` variable to determine available term buffers.
- Relay prefix argument to `multi-term` (to choose shell), if any.
- If no name is given when creating a new term buffer, default to current working directory of the launched process.
- As a consequence of ^^, remove `C-c n` keybinding and related  "auto-terminal" code.
- Wire `M-D` to delete marked items in Helm  buffer (same as `helm-buffer-list`).

Let me know what you think -- thanks!

fritzgrabo committed:
Simplify code by using functionality already provided by `multi-term`.

Note that this is somewhat opinionated as it drops support for creation
of terminals in `shell-mode`. I realize that this can be considered a
show-stopper by some. That said, if `shell-mode` support is desired,
shouldn't it rather go into `multi-term`?

- Use existing `multi-term-buffer-list` variable to determine available
  term buffers.
- Relay prefix argument to `multi-term` (to choose shell), if any.
- If no name is given when creating a new term buffer, default to
  current working directory.
- As a consequence of ^^, remove `C-c n` keybinding and related
  "auto-terminal" code.

dfdeshom commented:
I would rather keep the shell-mode functionality, I like it. The other changes seem ok, maybe re-submit them as a separate PR

fritzgrabo committed:
Fix `helm-mt/term-source-terminal-not-found`

fritzgrabo commented:
Thanks for the feedback. Sounds like I'll be rolling back most of my changes then, but that's ok -- I'm primarily using all of this to learn elisp anyway :). Will give it a go.

fritzgrabo committed:
Provide `M-D' shortcut to delete marked terms.

`helm-buffers-list` provides the same shortcut to delete marked buffers.

fritzgrabo committed:
Bring back `shell-mode`.

- Still stick to `multi-term-buffer-list` for terminals managed by
  `multi-term`.
- Hard-code available modes ('term and 'shell): they need custom logic
  anyway so no point in making them customizable?

fritzgrabo committed:
Update file header comments and README to reflect changes of previous commits.

fritzgrabo committed:
Bring back advice to reroute terminal functions to `helm-mt` when called interactively.

fritzgrabo commented:
Ready for re-review.

fritzgrabo committed:
Derive buffer name from working directory of launched process by
default.

This is required to avoid a mismatch between a terminal buffer's name
and its actual current working directory, which might differ from the
previously used `default-directory'.

An example of such a mismatch would be a multi-term that was advised by
projectile to launch in the current project's root directory rather than
in `default-directory`.

Note that this functionality depends on the `lsof` command (available on
both linux and osx platforms).

fritzgrabo committed:
Adjust to recent updates of the `helm` package

See https://github.com/emacs-helm/helm/commit/4be5c8 for details.

fritzgrabo commented:
Hi @dfdeshom, am curious: did you find some time to have a second look at this? Let me know if I can make it any better. I've been using this version for a month now, and it's been working well for me. Thanks!

dfdeshom commented:
@fritzgrabo thanks! Merged
